server/router/services/cluster-service.spec.js

```diff
@@ -19,110 +19,36 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-import clusterService from './cluster-service';
+import ClusterService from './cluster-service';
 
 describe('clusterService', () => {
-  describe('getCluster', () => {
-    beforeEach(() => {
-      clusterService.clearCache();
-    });
-
-    describe('cache = null', () => {
-      it('should return data from ctx.cadence.describeCluster.', async () => {
-        const cluster = {
-          persistenceInfo: {},
-          membershipInfo: null,
-          version: 1,
-        };
-
-        const ctx = {
-          cadence: {
-            describeCluster: jest.fn().mockImplementation(() => cluster),
-          },
-        };
-
-        const data = await clusterService.getCluster(ctx);
-
-        expect(data).toEqual(cluster);
-      });
-    });
-
-    describe('cache is set', () => {
-      it('should return cache on the second attempt.', async () => {
-        const clusterCache = {
-          persistenceInfo: {},
-          membershipInfo: null,
-          version: 1,
-        };
-
-        const ctx = {
-          cadence: {
-            describeCluster: jest.fn().mockImplementation(() => clusterCache),
-          },
-        };
-
-        const data = await clusterService.getCluster(ctx);
-
-        expect(data).toEqual(clusterCache);
-
-        const clusterUpdated = {
-          persistenceInfo: {},
-          membershipInfo: null,
-          version: 2,
-        };
+  let clusterService;
 
-        ctx.cadence.describeCluster.mockImplementation(() => clusterUpdated);
+  beforeEach(() => {
+    const cacheManagerMock = {
+      get: callback => callback(),
```

Comment:
what is `callback()` here?  I haven't been able to find it in mocha's docs...

Reply:
`CacheManager` is mocked here with a mocked `get` function. This is used by `ClusterService` in `getCluster` method. `callback` here would map directly to the `fetch` method in `ClusterService`. This would essentially mean that it ignores any caching logic and always call `fetch`.

Reply:
oh, derp.  I was misreading - this is just making an anonymous func that calls its argument.
makes sense!